src/main/java/org/assertj/core/api/AbstractPathAssert.java

```diff
@@ -0,0 +1,821 @@
+     * final Path notNormalized1 = fs.getPath("/a/./b");
+     * final Path notNormalized2 = fs.getPath("c/..");
+     *
+     * // the following assertions succeed:
+     * assertThat(normalized1).isNormalized();
+     * assertThat(normalized2).isNormalized();
+     * assertThat(normalized3).isNormalized();
+     * assertThat(normalized4).isNormalized();
+     *
+     * // the following assertions fail:
+     * assertThat(notNormalized1).isNormalized();
+     * assertThat(notNormalized2).isNormalized();
+     * </code></pre>
+     *
+     * @return self
+     */
+    public S isNormalized()
+    {
+        paths.assertIsNormalized(info, actual);
+        return myself;
+    }
+
+    /**
+     * Assert whether the tested path is canonical
+     *
+     * <p>For Windows users, this assertion is no different than {@link
+     * #isAbsolute()}. For Unix users, this assertion ensures that the tested
+     * path is the actual file system resource, that is, it is not a {@link
+     * Files#isSymbolicLink(Path) symbolic link} to the actual resource, even if
+     * the path is absolute.</p>
+     *
+     * <p>Example:</p>
+     *
+     * <pre><code class="java">
+     * // fs is a Unix filesystem
+     * // Create a directory
+     * final Path basedir = fs.getPath("/tmp/foo");
+     * Files.createDirectories(basedir);
+     * // Create a file in this directory
+     * final Path file = basedir.resolve("file");
+     * Files.createFile(file);
+     * // Create a symbolic link to that file
+     * final Path symlink = basedir.resolve("symlink");
+     * Files.createSymbolicLink(symlink, file);
+     *
+     * // The following assertion succeeds
+     * assertThat(file).isCanonical();
+     *
+     * // The following assertion fails
+     * assertThat(symlink).isCanonical();
+     * </code></pre>
+     *
+     * @throws PathsException an I/O error occurred while evaluating the path
+     *
+     * @see Path#toRealPath(LinkOption...)
+     * @see Files#isSameFile(Path, Path)
+     */
+    public S isCanonical()
+    {
+        paths.assertIsCanonical(info, actual);
+        return myself;
+    }
+
+    /**
+     * Assert whether the tested path has the expected parent path
+     *
+     * <p><em>This assertion will perform canonicalization of the tested path
+     * and of the given argument before performing the test; see the class
+     * description for more details. If this is not what you want, use {@link
+     * #hasParentRaw(Path)} instead.</em></p>
+     *
+     * <p>Checks that the tested path has the given parent. This assertion will
+     * fail both if the tested path has no parent, or has a different parent
+     * than what is expected.</p>
+     *
+     * <p>Examples:</p>
+     *
+     * <pre><code class="java">
+     * // fs is a Unix filesystem
+     * final Path actual = fs.getPath("/dir1/dir2/file");
+     * // this path will be normalized to "/dir1/dir2"
+     * final Path good = fs.getPath("/dir1/dir3/../dir2/.");
```

Comment:
add final Path normalized = fs.getPath("/dir1/dir2"); to show that it will pass the assertion